opentelemetry/src/sdk/metrics/mod.rs

```diff
@@ -79,9 +79,14 @@ struct MapKey {
 
 #[derive(Default, Debug)]
 struct AsyncInstrumentState {
-    /// runners maintains the set of runners in the order they were
-    /// registered.
-    runners: Vec<(AsyncRunner, Arc<dyn sdk_api::AsyncInstrumentCore>)>,
+    /// The set of runners in the order they were registered.
+    ///
+    /// Runners and instruments are 1:1 except batch observers, which have a
+    /// single runner for all their instruments.
+    runners: Vec<(AsyncRunner, Option<Arc<dyn sdk_api::AsyncInstrumentCore>>)>,
```

Comment:
This seems complex enough that it might be nice to define a separate type for it.

Reply:
extracted type alias for this